src/zope/app/server/tests/test_mkzopeinstance.py

```diff
@@ -18,42 +18,44 @@
 import sys
 import tempfile
 import unittest
-import zope.app.server
-
-from StringIO import StringIO
+from contextlib import contextmanager
 
 from zope.app.server import mkzopeinstance
 
+try:
+    from cStringIO import StringIO
+except ImportError:
+    from io import StringIO
```

Comment:
`io.BytesIO` on Python 2 to be sure we don't try to print unicode to stdout? This is just like zope.server.http.tests.test_wsgiserver (and I know I've seen/written this code somewhere else too...maybe it should be a standard part of zope.testing)
```python
# By using io.BytesIO() instead of cStringIO.StringIO() on Python 2 we make
# sure we're not trying to accidentally print unicode to stdout/stderr.
NativeStringIO = BytesIO if str is bytes else StringIO


@contextmanager
def capture_output(stdout=None, stderr=None):
    old_stdout = sys.stdout
    old_stderr = sys.stderr
    sys.stdout = stdout = stdout or NativeStringIO()
    sys.stderr = stderr = stderr or NativeStringIO()
    try:
        yield stdout, stderr
    finally:
        sys.stdout = old_stdout
        sys.stderr = old_stderr
```